### DISCUSSION TITLE
How to write composite key in migration, or, how to avoid overwrite existing generated entities

### DISCUSSION THREAD

**Discussion by boan-anbo** [SeaQL/sea-orm#1903] [Q&A] [answered]
Hi I have two questions, either of which can solve the problem I have right now: having to manually add attributes to generated entities.

Since I cannot add composite primary keys via migration (which raises more than one primary key exception), I have to manually add the `#[sea_orm(primary_key)]`.

And yet, we are prototyping right now and the entities get constantly regenerated and overwritten. 

So what's the solution to either generate the composite primary key, or do not let the cli to overwrite the existing entities?

```
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, Serialize, Deserialize)]
#[sea_orm(table_name = "document_annotation")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub annotation_id: i32,
    #[sea_orm(primary_key)]
    pub document_id: i32,
}
```

**Comment by boan-anbo** [comment, marked as answer]
I found the answer to how to write composite key migration on the discord:

https://discord.com/channels/873880840487206962/1084893969978249226/1084901054468071516

In short:

```rust
 .primary_key(
                        Index::create()
                            .col(UserPermission::UserId)
                            .col(UserPermission::PermissionId),
                    )
```

**Comment by tyt2y3** [comment]
Thanks. Just to point out these can be found in https://github.com/SeaQL/sea-orm/blob/master/tests/common/bakery_chain/schema.rs
